vertica_python/vertica/cursor.py

```diff
@@ -58,8 +58,7 @@ def execute(self, operation, parameters=None):
 
         self.rowcount = 0
         if self.last_execution:
-            # ToDo: can just empty the message buffer in connection if easy to do
-            self.connection.reset_connection()
```

Comment:
I agree that we need a general-purpose message handling function (the original version of this had a single function that dealt with everything), I think that will make the cursor implementation a lot cleaner.